packages/core/src/StateMachine.ts

```diff
@@ -446,58 +446,64 @@ export class StateMachine<
   }
 
   public restoreState(
-    state: State<TContext, TEvent, TResolvedTypesMeta>,
+    state: PersistedMachineState<any>,
     _actorCtx?: ActorContext<
       TEvent,
       State<TContext, TEvent, TResolvedTypesMeta>
     >
   ): State<TContext, TEvent, TResolvedTypesMeta> {
-    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
+    const self = this;
+    let restoredState;
 
-    if ('persisted' in state) {
-      const restoredChildren: Record<string, AnyActorRef> = {};
+    const children = {};
 
-      Object.keys(state.children).forEach((id) => {
-        const persistedState = state.children[id];
-        const impl = this.options.actors[id];
+    if (_actorCtx && state.persisted) {
```

Comment:
this `state.persisted` check is redundant now since you now require `PersistedMachineState` as the argument

Reply:
this also means that the `else` branch could be removed from here